edn_format/immutable_list.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import collections
+
+
+class ImmutableList(collections.MutableSequence, collections.Hashable):
+    def __init__(self, wrapped_list, copy=True):
+        self._list = list(wrapped_list) if copy else wrapped_list
```

Comment:
Good question, @bfontaine .

@konr I guess this is okay, since we don't want to reinvent the wheel here. The ideal solution may be something like https://github.com/tobgu/pyrsistent but that's a big change...